### DISCUSSION TITLE
Cant Figure Out Multiple PC Monitoring

### DISCUSSION THREAD

**Discussion by Lukemul6** [h2oai/wave#1865] [Q&A]
I am having trouble figuring out how to link multiple PCs up that are on the same network into one screen. I want to be able to monitor the CPU usage of the 2 computers, I have tried looking around but there isnt much information that I have found. 

If someone can point me in the right direction? 

Many thanks!

**Comment by mturoci** [comment]
Hi @Lukemul6.

> how to link multiple PCs up that are on the same network into one screen

What is the definition of a screen here? Is it a Wave app/script?

You can either:

* Have all your PCs write to a shared database which would be polled by your Wave app and display the results. The latency might be an issue here.
* If you just want to display the dashboard without capturing user input, [wave scripts](https://wave.h2o.ai/docs/scripts/) are enough in that case. All you need to do then is to:

1. Start wave server (waved) on whichever PC. (`./waved`)
2. For each PC, setup a periodically run python script (e.g. cron job), that would update the content on your page.

For example script for PC1 (script1) could be

```py
from h2o_wave import site, ui

# Get the web page at route '/'.
# If you're running this example on your local machine,
# this page will refer to http://localhost:10101/.
page = site['/']

# Update CPU usage on the page.
page['cpu1'] = ui.markdown_card(
    box='1 1 2 2',
    title='CPU Usage 1',
    content='...',
)

# Finally, sync the page to send our changes to the server.
page.save()
```

and for PC2 (script2.py):

```py
from h2o_wave import site, ui

# Get the web page at route '/'.
# If you're running this example on your local machine,
# this page will refer to http://localhost:10101/.
page = site['/']

# Update CPU usage on the page.
page['cpu2'] = ui.markdown_card(
    box='3 3 2 2',
    title='CPU Usage 2',
    content='...',
)

# Finally, sync the page to send our changes to the server.
page.save()
```

Then set the env variable `H2O_WAVE_ADDRESS` (see [docs](https://wave.h2o.ai/docs/configuration/#h2o_wave_address)) to point to your PC with `waved` running and run the scripts:

```sh
H2O_WAVE_ADDRESS='http://<ip-of-your-waved-machine>:port' python script1.py
```

For a more realtime-help with questions like these, I recommend joining our [Discord](https://discord.gg/V8GZFAy3WM).

**Comment by Lukemul6** [comment]
Thanks for your response. I made a post in the discord as I need further help.

**Comment by mturoci** [comment]
The discussion continues in https://discord.com/channels/1022255416454946906/1078877598119100546/1078877598119100546.
